proj1a/ArrayDeque.java

```diff
@@ -0,0 +1,185 @@
+public class ArrayDeque<T> implements Deque<T> {
+    private T[] items;
+    private int size;
+    private int nextFirst;
+    private int nextLast;
+    private int RFACTOR;
+    private int R;
```

Comment:
Why do we need this instance variable?

Reply:
This is used to represent usage ratio of the array. However, this should not be an instance for the ArrayDeque class. I have put the definition of this variable into the corresponding method where it was used.

Reply:
Yes, it should be a local variable